questionnaires/views.py

```diff
@@ -20,3 +41,143 @@ def get_filename(self):
 
     def get_queryset(self):
         return super().get_queryset().select_related('page', 'user')
+
+
+class QuestionnairesView(SpreadsheetExportMixin, SafePaginateListView):
+    template_name = "questionnaires/form_data.html"
+    context_object_name = 'form_pages'
+    list_export = ['ID', 'Name', 'Submission Date', 'Field', 'Value']
+    select_date_form = SelectDateForm
+    page_ids = []
+    user = None
+
+    def dispatch(self, request, *args, **kwargs):
+        self.is_export = (self.request.GET.get('export') in self.FORMATS)
+        user_id = self.request.GET.get('user_id')
+        self.user = get_object_or_404(User, pk=user_id)
+        if self.is_export:
+            page_ids = self.request.GET.get('page_ids')
+            self.page_ids = page_ids.split(',') if page_ids else []
+            self.paginate_by = None
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_queryset(self):
+        """ Return the queryset of form pages for this view """
+        queryset = get_forms_for_user(self.request.user)
+        queryset = queryset.filter(
+            usersubmission__user_id=self.user.id
+        ).distinct().order_by('-last_published_at')
+
+        filtering = self.get_filtering()
+        if filtering and isinstance(filtering, dict):
+            queryset = queryset.filter(**filtering)
+
+        return queryset
+
+    def get_filtering(self, for_form_pages=True):
+        """ Return filering as a dict for form pages or submissions queryset """
+        filter_name = 'usersubmission__submit_time' if for_form_pages else 'submit_time'
+        self.select_date_form = SelectDateForm(self.request.GET)
+        result = dict()
+        if self.select_date_form.is_valid():
+            date_from = self.select_date_form.cleaned_data.get('date_from')
+            date_to = self.select_date_form.cleaned_data.get('date_to')
+            if date_to:
+                # careful: date_to must be increased by 1 day
```

Comment:
if you agree, please change this comment to:
```
# date_to is advanced by 1 day to be inclusive of date_to in results
```

Reply:
used date filter instead of datetime, so don't need this comment now